### PR TITLE
make subcommands return non-zero status on failures

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/2518

Makes subcommands return non-zero status on failures.  Also creates `osc policy` for handling non-cluster admin policy commands.

### CODE REVIEW

**Comment by deads2k** [comment]
@liggitt

File: pkg/cmd/admin/policy/policy.go [hunk 1]
```diff
@@ -47,6 +48,9 @@ func NewCommandPolicy(name, fullName string, f *clientcmd.Factory, out io.Writer
 
 func runHelp(cmd *cobra.Command, args []string) {
 	cmd.Help()
+
+	// make sure that we exit with non-zero status so that typoed commands don't look like they were successful
+	os.Exit(1)
```
Comment (liggitt) [inline pkg/cmd/admin/policy/policy.go]: what an exciting thing to find in such an innocuously named function :)

File: pkg/cmd/admin/admin.go [hunk 2]
```diff
@@ -37,6 +38,8 @@ func NewCommandAdmin(name, fullName string, out io.Writer) *cobra.Command {
 		Run: func(c *cobra.Command, args []string) {
 			c.SetOutput(out)
 			c.Help()
+
+			os.Exit(1)
```
Comment (liggitt) [inline pkg/cmd/admin/admin.go]: does this make running `openshift admin` return `1` as well?

File: pkg/cmd/admin/policy/policy.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/admin/policy/policy.go]: same question here, does this return `1` in cases where you just ran the base command?

File: pkg/cmd/admin/admin.go (hunk 2 above)
Comment (deads2k) [inline pkg/cmd/admin/admin.go]: Is running a non-runnable command ever anything besides an error?

**Comment by deads2k** [comment]
Only non-zero when args > 0.

**Comment by deads2k** [comment]
@liggitt bump

**Comment by mfojtik** [comment]
LGTM

**Comment by liggitt** [comment]
if we only return error statuses for base commands when they get an unexpected arg (meaning a child command got typoed, etc), I'm ok with that

**Comment by deads2k** [comment]
[merge]

**Pull request merged**